Backend/src/controller/stock-price.js

```diff
@@ -2,26 +2,53 @@ import dotenv from "dotenv";
 import "isomorphic-fetch"
 dotenv.config({ silent: process.env.NODE_ENV === 'production' });
 
+function getStockURL(stock) {
+	return "https://www.alphavantage.co/query?function=GLOBAL_QUOTE&symbol="+
+			stock + "&apikey=" + process.env.ALPHAVANTAGE_KEY;
+}
+
+async function getStockPrices(stocks) {
+	let stockPrices = new Map();
+	const fetchCalls = stocks.map(stock => {
+		console.log("stockPrices");
+		const url = getStockURL(stock);
+		console.log(url);
+    	fetch(url)
+    		.then(response => response.json())
+    		.then(data => {
+		  		const price = data["Global Quote"]["05. price"];
+		  		stockPrices[stock] = price; })
+    		.catch((error) => {
+  				console.log("Can't find price for: " + stock + ".");
+			});
+		}
+  	);
+	await Promise.all(fetchCalls);
```

Comment:
This is currently not working. Do you know how to sue async and await with promises? @ngupta-iwasaki 

Currently it is returning `Promise { <pending> }`, but I would like it to return the value whenever the fetches are complete.

Reply:
I'll take a look

Reply:
Promise.all is resolving immediately because fetchCalls is not returning anything

Reply:
Is this really how this works? You have two return statements?

Reply:
> Is this really how this works? You have two return statements?

Nvm, the first return statement in within an inner function.